paf-mvp-benchmark/OpenTelemetry/README.md

```diff
@@ -77,3 +77,29 @@ All services inherited from class Node are monitored.
 ### Viewers
 A docker image with a collector, Jaeger and Prometheus is given under paf-mvp-benchmark.\
 Default viewer is accessible at [Jaeger UI](http://localhost:16686/search).
+
+
+## Span correlation
+[Spans](https://opentelemetry.io/docs/concepts/signals/traces/#spans-in-opentelemetry) are now started in the `beginHandling` handler and ended in the `endHandling` of each endpoint.\
+Response object is used to store and keep track of the current span.\
+With the current implementation we have the guarantee that **spans** within the **same endpoint** are correctly correlated but that will not be the case when chaining requests (ex: pafLib => clientNode => operatorNode).\
+To do so, we need to make use of `correlation-id` propagated in the http header of each request. For now, we added this `correlation-id` as a span attribute to be able to visualize it on the **jaeger-ui** and correlate spans by using manual filtering.\
+A more sophisticated solution, would be to make use of **context propagation**. To do so we need to:
+1. Identify request chains (paf, clientNode, operatorNode)
+2. Create custom contexts: A **context** is basically a key-value store. we can define a `correlation-id` key on the defaultContext like this:
+```` 
```

Comment:
[NIT] precise the language to have GitHub to do syntax highlighting:
` ```typescript`

Reply:
Good point! thank you